phplib/HTML/QuickForm.php

```diff
@@ -384,7 +384,7 @@ function registerElementType($typeName, $include, $className)
     function registerRule($ruleName, $type, $data1, $data2 = null)
     {
         include_once( preg_replace('/HTML\/QuickForm.*$/', '', __FILE__ ) . 'HTML/QuickForm/RuleRegistry.php');
-        $registry =& HTML_QuickForm_RuleRegistry::singleton();
+        $registry = (new HTML_QuickForm_RuleRegistry)->singleton();
```

Comment:
I think this commit's object change could be done by making `singleton` a static function instead

Reply:
I've reverted this change and made the `singleton` function a static, tests still pass and it all looks good I think.